pkg/api/types.go

```diff
@@ -1753,6 +1753,36 @@ type ReplicationControllerStatus struct {
 
 	// ObservedGeneration is the most recent generation observed by the controller.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Represents the latest available observations of a replication controller's current state.
+	Conditions []ReplicationControllerCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type ReplicationControllerConditionType string
+
+// These are valid conditions of a replication controller.
+const (
+	// ReplicationControllerReplicaFailure is added in a replication controller when one of its pods
+	// fails to be created or deleted.
```

Comment:
https://github.com/kubernetes/kubernetes/issues/32863#issuecomment-249075791

Eventually I want to surface ImagePull errors and crashlooping pods under this (or maybe a new one) Condition too.